iceoryx_hoofs/include/iceoryx_hoofs/cxx/span_iterator.hpp

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#ifndef IOX_HOOFS_CXX_SPAN_ITERATOR_HPP
+#define IOX_HOOFS_CXX_SPAN_ITERATOR_HPP
+
+#include <iterator> // for reverse_iterator, distance, random_access_...
+
+#include "iceoryx_hoofs/cxx/requires.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+
+
+inline constexpr bool ConstexprCheckTrue(bool condition)
```

Comment:
@patdhlk 

- [ ] create ExpectsWithMsg variant

Reply:
Added a comment to refactor this check with an `if constexpr` once moving to C++17.